test/unit_test/static/unit_test.cpp

```diff
@@ -46,104 +50,371 @@ inline constexpr struct newton_ : named_unit<"N", kilogram * metre / square<seco
+static_assert(is_of_type<1 / (second * second), derived_unit<one_, per<power<second_, 2>>>>);
+static_assert(is_of_type<1 / (1 / (second * second)), derived_unit<power<second_, 2>>>);
+
+static_assert(is_of_type<metre / second * (1 / second), derived_unit<metre_, per<power<second_, 2>>>>);
+static_assert(is_of_type<metre / second*(metre / second), derived_unit<power<metre_, 2>, per<power<second_, 2>>>>);
+static_assert(is_of_type<metre / second*(second / metre), one_>);
 
-static_assert(kilo<metre> == kilometre);
-static_assert(mag<1000> * metre == kilo<metre>);
+static_assert(is_of_type<watt / joule, derived_unit<watt_, per<joule_>>>);
+static_assert(is_of_type<joule / watt, derived_unit<joule_, per<watt_>>>);
+
+
+// derived unit normalization
+constexpr auto u4 = metre / second;
+static_assert(is_of_type<get_canonical_unit(u4).reference_unit, derived_unit<metre_, per<second_>>>);
+static_assert(get_canonical_unit(u4).mag == mag<1>);
+
+constexpr auto u5 = kilometre / second;
+static_assert(is_of_type<u5, derived_unit<kilometre_, per<second_>>>);
+static_assert(is_of_type<get_canonical_unit(u5).reference_unit, derived_unit<metre_, per<second_>>>);
+static_assert(get_canonical_unit(u5).mag == mag<1000>);
+
+constexpr auto u6 = kilometre / hour;
+static_assert(is_of_type<u6, derived_unit<kilometre_, per<hour_>>>);
+static_assert(is_of_type<get_canonical_unit(u6).reference_unit, derived_unit<metre_, per<second_>>>);
+static_assert(get_canonical_unit(u6).mag == mag<ratio{1000, 3600}>);
+
+constexpr auto u7 = mag<1000> * kilometre / hour;
+static_assert(is_of_type<u7, derived_unit<scaled_unit<mag<1000>, kilometre_>, per<hour_>>>);
+static_assert(is_of_type<get_canonical_unit(u7).reference_unit, derived_unit<metre_, per<second_>>>);
+static_assert(get_canonical_unit(u7).mag == mag<ratio{1'000'000, 3'600}>);
+
+constexpr auto u8 = mag<1000> * (kilometre / hour);
+static_assert(is_of_type<u8, scaled_unit<mag<1000>, derived_unit<kilometre_, per<hour_>>>>);
+static_assert(is_of_type<get_canonical_unit(u8).reference_unit, derived_unit<metre_, per<second_>>>);
+static_assert(get_canonical_unit(u8).mag == mag<ratio{1'000'000, 3'600}>);
+
+constexpr auto u9 = 1 / hour * (mag<1000> * kilometre);
+static_assert(is_of_type<u9, derived_unit<scaled_unit<mag<1000>, kilometre_>, per<hour_>>>);
+static_assert(is_of_type<get_canonical_unit(u9).reference_unit, derived_unit<metre_, per<second_>>>);
+static_assert(get_canonical_unit(u9).mag == mag<ratio{1'000'000, 3'600}>);
+
+// comparisons of the same units
+static_assert(second == second);
+static_assert(metre / second == metre / second);
+
+// comparisons of equivalent units (named vs unnamed/derived)
+static_assert(1 / second == hertz);
+static_assert(convertible(1 / second, hertz));
+
+// comparisons of equivalent but not convertible units
+static_assert(hertz == becquerel);
+static_assert(convertible(hertz, becquerel));
```

Comment:
The comment and the code don't match here. Going by the comment, `convertible` is missing `not` in front of it.

Reply:
Right, old comment. Thanks! Initially they were not meant to be convertible but I was unable to find a way to code it and make everything else work fine 😢